physics/clm_lake.f90

```diff
@@ -0,0 +1,5607 @@
+!> \file clm_lake.F90
+!!  Contains code related to the CLM lake model
+!!
+!! This lake scheme was taken from module_sf_lake in WRF 4.3.1, and
+!! modified for CCPP by Sam Trahan in June 2022.
+!!
+!! The original documentation said:
+!! 
+!! The lake scheme was retrieved from the Community Land Model version 4.5 
+!! (Oleson et al. 2013) with some modifications by Gu et al. (2013). It is a 
+!! one-dimensional mass and energy balance scheme with 20-25 model layers, 
+!! including up to 5 snow layers on the lake ice, 10 water layers, and 10 soil 
+!! layers on the lake bottom. The lake scheme is used with actual lake points and 
+!! lake depth derived from the WPS, and it also can be used with user defined 
+!! lake points and lake depth in WRF (lake_min_elev and lakedepth_default). 
+!! The lake scheme is independent of a land surface scheme and therefore 
+!! can be used with any land surface scheme embedded in WRF. The lake scheme 
+!! developments and evaluations were included in Subin et al. (2012) and Gu et al. (2013) 
+!!
+!!   Subin et al. 2012: Improved lake model for climate simulations, J. Adv. Model. 
+!!
+!!   Earth Syst., 4, M02001. DOI:10.1029/2011MS000072; 
+!!
+!!   Gu et al. 2013: Calibration and validation of lake surface temperature simulations 
+!!
+!!   with the coupled WRF-Lake model. Climatic Change, 1-13, 10.1007/s10584-013-0978-y. 
+
+MODULE clm_lake
+  
+    use machine,               only: kind_phys, kind_dbl_prec
+
+    implicit none 
+
+    private
+
+    public :: clm_lake_run, clm_lake_init, LAKEDEBUG
+
+    integer, parameter, public :: kind_lake = kind_dbl_prec
```

Comment:
Why does CLM need its own real precision? I.e. Why can't it use kind_phys to be controlled like the rest of the physics?

Reply:
It was double precision in the WRF, and we're uncertain of whether it needs to be double precision. For safety's sake, we're leaving it as double precision for now. The `kind_lake` lets us experiment with `kind_lake=kind_phys` without having to rewrite the whole file.

Reply:
I've added a comment explaining this.